rllib/core/models/catalog.py

```diff
@@ -0,0 +1,227 @@
+import gymnasium as gym
+from ray.rllib.core.models.configs import (
+    MLPEncoderConfig,
+    LSTMEncoderConfig,
+    CNNEncoderConfig,
+)
+from ray.rllib.core.models.base import ModelConfig
+from ray.rllib.models import MODEL_DEFAULTS
+from gymnasium.spaces import Box
+from ray.rllib.models.utils import get_filter_config
+
+
+class Catalog:
+    """Defines how what models an RLModules builds.
+
+    RLlib's native RLModules get their Models from a Catalog object.
+    By default, that Catalog builds the configs it holds.
+    You can modify a Catalog so that it builds different Models by subclassing and
+    overriding the build_* methods. Alternatively, you can customize the configs
+    inside RLlib's Catalogs to customize what is being built by RLlib.
+
+    Usage example:
+
+    # Define a custom catalog
+
+    .. testcode::
+
+    class MyCatalog(Catalog):
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        model_config_dict: dict,
+    ):
+        super().__init__(observation_space, action_space, model_config_dict)
+        self.my_model_config = MLPModelConfig(
+            hidden_layer_dims=[64, 32],
+            input_dim=self.observation_space.shape[0],
+            output_dim=1,
+        )
+
+        def build_my_head(self, framework: str):
+            return self.my_model_config.build(framework=framework)
+
+
+    # With that, RLlib can build and use models from this catalog like so:
+    catalog = MyCatalog(gym.spaces.Box(0, 1), gym.spaces.Box(0, 1), {})
+    my_head = catalog.build_my_head("torch")  # doctest: +SKIP
+    out = my_head(...)  # doctest: +SKIP
+
+    # We can also modify configs of RLlib's native Catalogs like so:
+    catalog = MyCatalog(gym.spaces.Box(0, 1), gym.spaces.Box(0, 1), {})
+    catalog.my_model_config.output_dim = 32
+    my_head = catalog.build_my_head("torch")  # doctest: +SKIP
+    out = my_head(...)  # doctest: +SKIP
+
+    """
+
+    def __init__(
+        self,
+        observation_space: gym.Space,
+        action_space: gym.Space,
+        # TODO (Artur): Turn model_config into model_config_dict to distinguish
```

Comment:
I thought I should try to separate PRs by concerns. I thought it would have nothing to do with the exact PR and would clutter it. I'm changing it nonetheless now since you will re-review anyway and know about this now 🙂

Reply:
I thought I'd do it library wide, but I think that time will come closer to when we flip the switch.
Changed this in the rllib/core/models/ folder now and left a TODO for the rest.